segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -1694,7 +1695,7 @@ private void write00(SegmentSnapshotRecord object, RevisionDataOutput output) th
 
             private void read00(RevisionDataInput input, SegmentSnapshotRecord.SegmentSnapshotRecordBuilder b) throws IOException {
                 b.segmentMetadata((SegmentMetadata) SEGMENT_METADATA_SERIALIZER.deserialize(input.getBaseStream()));
-                b.chunkMetadataCollection(input.readCollection(ELEMENT_DESERIALIZER));
+                b.chunkMetadataCollection(input.readCollection(ELEMENT_DESERIALIZER, () -> new Vector<>()));
```

Comment:
Why is this needed? Is `chunkMetadataCollection` modified after the object is built?

Reply:
same as above

Reply:
Ok. Great. Then let's use: ImmutableList and use ImmutableList.builder() it will make this explicit and have higher performance